pkg/workflows/sdk/v2/promise.go

```diff
@@ -0,0 +1,58 @@
+package sdk
+
+import (
+	"sync"
+)
+
+type Promise[T any] interface {
+	Await() (T, error)
+	CapabilityPromise
+}
+
+type CapabilityPromise interface {
+	promise()
+}
+
+func NewBasicPromise[T any](await func() (T, error)) Promise[T] {
+	return &basicPromise[T]{await: await}
+}
+
+func PromiseFromResult[T any](result T, err error) Promise[T] {
+	return &basicPromise[T]{resolved: result, err: err, isResolved: true}
+}
+
+type basicPromise[T any] struct {
```

Comment:
Does this behave differently than [sync.OnceValues](https://pkg.go.dev/sync#OnceValues)? It seems like we could leverage that, or even just expect callers to use it directly instead.